### ISSUE TITLE
How to end the hook function?

### ISSUE THREAD

**Issue by KeineLimonade** [IronManMark20/pyhooked#21]
First of all I love this small library.
But when I would llike to know how to terminate it´s execution.

For example I want to end the hook when the key A is pressed:

```
    self.hk = Hook()
    self.hk.handler = self.handle_events
    self.hk.hook()
def handle_events(self, args):
    if(self.ready):
        if isinstance(args, KeyboardEvent):
            print(args.key_code)
            if args.current_key == 'Left':
                self.callback(key.left)
            if args.current_key == 'A': 
                print("A")
                ENDHOOKING
```

So how do I do this?

**Comment by emmatyping** [comment]
There are two functions `unhook_mouse` and `unhook_keyboard` which will end the `hook` function and Python should garbage collect the `Hook` class as it is no longer used.

It might be a good idea to have one unified unhook function.

PS. Glad you like the library :smiley:

**Comment by KeineLimonade** [comment]
Well, using the `unhook_keyboard` function will only prevent the hook function from calling the eventhandeler.

In this code "ended" will never be printed:

```
from pyhooked import Hook, KeyboardEvent, MouseEvent
def handle_events(args):
    if isinstance(args, KeyboardEvent):
        print(args.key_code)
        if args.current_key == 'A' and args.event_type == 'key down' and 'Lcontrol' in args.pressed_key:
            print("Ctrl + A was pressed")
        if args.current_key == 'H' and args.event_type == 'key down' and 'Lcontrol' in args.pressed_key:
            hk.unhook_keyboard()

hk = Hook()
hk.handler = handle_events
hk.hook(keyboard=True)
print("ended")
```

**Comment by emmatyping** [comment]
Ah, interesting, I will look into how I can get it to unhook properly. Thanks for reporting this.

**Comment by chrisnorman7** [comment]
Hi,
I believe I have fixed this with #23.

**Comment by emmatyping** [comment]
With #23, there is now a `stop` function that ends the loop. Hope this meets your needs!
